csharp/Common/Exception/ErrorMessage.cs

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+using System;
+
+namespace TypeDB.Driver.Common
+{
+    public abstract class ErrorMessage
+    {
+        private readonly string _codePrefix;
```

Comment:
(duplicate old comment as it was outdated within the refactoring):

readonly vs { get; }:

*If I expect to always get the same result for each call, it's a readonly field.
*If I expect that the result can be different, it is a property with a getter (with no setters).
*If we call a native function under the hood, it's a property with a getter (with no setters).